linera-execution/src/evm/revm.rs

```diff
@@ -484,35 +487,49 @@ fn address_to_user_application_id(address: Address) -> ApplicationId {
 enum BaseRuntimePrecompile {
     /// Calling `chain_id` of `BaseRuntime`
     ChainId,
+    /// Calling `block_height_id` of `BaseRuntime`
+    BlockHeight,
     /// Calling `application_creator_chain_id` of `BaseRuntime`
     ApplicationCreatorChainId,
+    /// Calling `read_system_timestamp` of `BaseRuntime`
+    ReadSystemTimestamp,
+    /// Calling `read_chain_balance` of `BaseRuntime`
+    ReadChainBalance,
+    /// Calling `read_owner_balance` of `BaseRuntime`
+    ReadOwnerBalance(AccountOwner),
+    /// Calling `read_owner_balances` of `BaseRuntime`
+    ReadOwnerBalances,
+    /// Calling `read_balance_owners` of `BaseRuntime`
+    ReadBalanceOwners,
     /// Calling `chain_ownership` of `BaseRuntime`
     ChainOwnership,
     /// Calling `read_data_blob` of `BaseRuntime`
-    ReadDataBlob { hash: CryptoHash },
+    ReadDataBlob(CryptoHash),
     /// Calling `assert_data_blob_exists` of `BaseRuntime`
-    AssertDataBlobExists { hash: CryptoHash },
+    AssertDataBlobExists(CryptoHash),
 }
 
 /// Some functionalities from the ContractRuntime not in BaseRuntime
 #[derive(Debug, Serialize, Deserialize)]
 enum ContractRuntimePrecompile {
-    /// Calling `try_call_application` of `ContractRuntime`
-    TryCallApplication {
-        target: ApplicationId,
-        argument: Vec<u8>,
-    },
-    /// Calling `validation_round` of `ContractRuntime`
-    ValidationRound,
+    /// Calling `authenticated_signer` of `ContractRuntime`
+    AuthenticatedSigner,
```

Comment:
Isn't this the same as `tx.origin`?

Reply:
Not if you have a Wasm application that calls the EVM contract.
And in any case, the `tx.origin` is not set right now, but will be done later.